### PR TITLE
IO URL handling

### PR DESCRIPTION
**Pull request by InvalidPointer**
Implemented URL handling and added 2 tests for this functionality.

Waiting for feedback!

### CODE REVIEW

**Comment by InvalidPointer** [comment]
Travis build fails because of this https://github.com/cclib/cclib/issues/350

File: src/scripts/ccget [hunk 1]
```diff
@@ -90,15 +91,23 @@ def ccget():
         print(MSG_USAGE)
         sys.exit(1)
 
-    # Figure out which are the attribute names and which are the filenames.
+    # Figure out which are the attribute names and which are the filenames or links.
     # Note that in Linux, the shell expands wild cards, but not so in Windows,
     # so try to do that here using glob.
+    link_regex = re.compile(
```
Comment (langner) [inline src/scripts/ccget]: Can you re-use the constant from ccio?

File: src/cclib/io/ccio.py [hunk 2]
```diff
@@ -140,17 +150,64 @@ def ccopen(source, *args, **kargs):
     inputfile = None
     is_stream = False
 
+    link_regex = re.compile(
```
Comment (langner) [inline src/cclib/io/ccio.py]: Can you make this a constant?

File: src/scripts/ccget (hunk 1 above)
Comment (InvalidPointer) [inline src/scripts/ccget]: Done (https://github.com/cclib/cclib/pull/351/commits/344d3ee33587c6294db22855f068b7cc6ae4f215).

File: test/io/testccio.py [hunk 3]
```diff
@@ -14,6 +14,11 @@
 import cclib
 
 
+__filedir__ = os.path.dirname(__file__)
```
Comment (langner) [inline test/io/testccio.py]: Could you make a helper function for this?

File: test/io/testccio.py [hunk 4]
```diff
@@ -70,6 +75,26 @@ def test_cjson_empty_tempfile(self):
         with tempfile.NamedTemporaryFile() as tf:
             self.assertIsInstance(self.ccopen(tf.name, cjson=True), cclib.io.cjsonreader.CJSON)
 
+    def test_url_io(self):
+        """Does the function works with URLs such good as with filenames?"""
```
Comment (langner) [inline test/io/testccio.py]: Could you add a link to this issue?

**Comment by langner** [comment]
This looks great! Let's just wait for #355, so that we have clean tests before submitting this.

**Comment by InvalidPointer** [comment]
Thanks!

**Comment by InvalidPointer** [comment]
Now fails with:

ERROR: test_multi_url_io (testccio.ccopenTest)
Does the function works with multiple URLs such good as with multiple filenames?
----------------------------------------------------------------------
Traceback (most recent call last):
  File "io/testccio.py", line 102, in test_multi_url_io
    self.ccopen([os.path.join(fpath, fname) for fname in filenames], quiet=True)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/cclib/parser/logfileparser.py", line 339, in parse
    data = self.datatype(attributes=self.__dict__)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/cclib/parser/data.py", line 312, in __init__
    super(ccData_optdone_bool, self).__init__(*args, **kwargs)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/cclib/parser/data.py", line 180, in __init__
    self.setattributes(attributes)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/cclib/parser/data.py", line 315, in setattributes
    invalid = super(ccData_optdone_bool, self).setattributes(*args, **kwargs)
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/cclib/parser/data.py", line 252, in setattributes
    self.arrayify()
  File "/home/travis/virtualenv/python2.7.12/lib/python2.7/site-packages/cclib/parser/data.py", line 208, in arrayify
    setattr(self, k, numpy.array(getattr(self, k), precision))
ValueError: setting an array element with a sequence.

I'll study it deeper tomorrow.

**Comment by langner** [comment]
If it wouldn't be too much trouble, could you rebase your changes into fewer commits? Please just leave a few commits describing significant changes (no typo commits). Here are some instructions: https://help.github.com/articles/using-git-rebase-on-the-command-line/

**Comment by InvalidPointer** [comment]
Successfully reduced number of commits to 2, thanks @langner!

**Pull request merged**